google/resource_bigquery_table.go

```diff
@@ -93,6 +93,33 @@ func resourceBigQueryTable() *schema.Resource {
 				},
 			},
 
+			"field": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"mode": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "NULLABLE",
+							ValidateFunc: validation.StringInSlice([]string{"NULLABLE", "REQUIRED", "REPEATED"}, false),
```

Comment:
@danawillow this API doesn't really abstract away the underlying BQ API – should it? Basically, a field can _either_ be optional, required, or repeated. But maybe using a string is bad?

Reply:
Nah, this is how we usually do enum-like values.